src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -150,6 +154,27 @@ public async Task<Error> SaveChanges(Config config)
                 return null;
             }
 
+            var remainingRetries = 30;
+            var (error, response) = await SaveChangesCore(config, _etag);
+            while (error == null && remainingRetries-- > 0 && response.StatusCode == HttpStatusCode.PreconditionFailed)
+            {
+                try
+                {
+                    response = await HttpClientUtility.GetAsync(_url, config);
+                }
+                catch (HttpRequestException ex)
```

Comment:
can this and [here](https://github.com/dotnet/docfx/blob/v3/src/docfx/restore/RestoreFile.cs#L108) be `catch (Exception)` because `GetAsync` may throw other exception types as well.